### PR TITLE
Migrating Code to use Proper Byond Move() Code

### PR DESCRIPTION
**Pull request by clusterfack**
The goal of this code is to cause the proper use of Cross() and Uncross() in the code and remove it all from turf/Enter()

This primarily involves refactoring code that already exists (such as Canpass and CheckExit) into more friendly code.

### CODE REVIEW

**Comment by Exxion** [comment]
Well this will make my super secret project that probably won't actually happen a lot easier

**Comment by clusterfack** [comment]
Cool

**Comment by Exxion** [comment]
BYOND's docs never get into special cases, but do you happen to know if our snowflake border object code could, with this, be replaced with standard dense objects with `bound_width` or `bound_height` set to 0 (and the other left at the default)?

**Comment by clusterfack** [comment]
If you're talking about implementing pixel movement then yes it would be much easier to handle border objects. This also so insanely simplifies turf/enter pixel movement would be SIGNIFICANTLY easier.

**Comment by clusterfack** [comment]
Note: None of the turf/enter code at this stage has been rewritten yet.

**Comment by Exxion** [comment]
Yes, implementing pixel movement later would be neat, but I just meant to ask if after this PR is done but we're still on tilelocked movement, BYOND would recognize `Cross()` on a zero-width/height bounding box

**Comment by clusterfack** [comment]
Not sure if I understand your question, I'm not sure what properties a zero height/width bounding box has but it should still have a loc. In tilebased movement cross() is called when you enter a tile for all atom/movables in the loc.

**Comment by Exxion** [comment]
Oh, I didn't realize that behavior of things other than gliding changed based on whether or not step-based movement is used

Also yes you guessed my secret project correctly
Making `Adjacent()` not awful without significantly changing its behavior in non-tilebased movement is my current conundrum
Fuck obstructions

**Comment by clusterfack** [comment]
Not going to lie pixel movement would be HYPE, besides client dir and fps raising and optimizations, it was the only other major project I ever wanted to see in ss13.

**Comment by clusterfack** [comment]
(FPS raising was actually my goal as a stepping stone to pixel movement because pixel movement looks like shit at low FPS)

**Comment by Exxion** [comment]
I actually just started thinking about that on... Thursday, I think, but at this point I don't see the point in working on it until after this PR is done and merged
After this PR is one and merged, I think the hardest parts of implementing pixel movement would be finding and eliminating unnecessary snowflake code and fixing `Adjacent()`

**Comment by clusterfack** [comment]
Oh no no, your hardest part will be replacing portions of the code that use loc for what are single tile objects right now, to locs for what will now be multitile objects.

**Comment by Exxion** [comment]
OK yeah that too
Forgot about that

**Comment by Exxion** [comment]
RIP

**Comment by Exxion** [comment]
@clusterfack 
If you don't want to finish this, I would "happily" take over
This seems like just the kind of project that I would absolutely hate but also derive some kind of masochistic pleasure from completing

**Comment by clusterfack** [comment]
This is very close to being completed

**Comment by Exxion** [comment]
Oh nice
I figured you stopped because no commits had been pushed in quite a long time but I guess not

**Comment by PJB3005** [comment]
I think it's more that you made him remember this still exists.

**Comment by clusterfack** [comment]
It's actually that I ran into some really annoying logistical questions of how to get this to work properly, and only after pondering for this long have I stumbled upon the correct answers

**Comment by clusterfack** [comment]
All donesies

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -143,28 +143,10 @@
...
-					if (step(src, SOUTH))
-						. = step(src, EAST)
-					else if (step(src, EAST))
-						. = step(src, SOUTH)
-				else if (Dir & 8)
-					if (step(src, SOUTH))
-						. = step(src, WEST)
-					else if (step(src, WEST))
-						. = step(src, SOUTH)
+			if (step(src, Dir & NORTH|SOUTH))
```
Comment (clusterfack) [inline code/game/atoms_movable.dm]: Don't you think this looks prettier? I do

**Comment by clusterfack** [comment]
I think Cross() is called in order of layer from highest to lowest as well so it prevents you from hitting a powered grille when it is covered by a window

**Comment by clusterfack** [comment]
Separating out ZAS' height/air_group from Cross() would probably be beneficial in the future for performance reasons, but I didn't feel it was necessary to include this time because I couldn't be assured it wouldn't break.

**Comment by CptWad** [comment]
this broke shit I think
mice can fly through walls 
wheelchairs can too
various other weird bugs too see #9849

**Comment by Exxion** [comment]
WHOP WHOP

**Comment by gbasood** [comment]
broken shit fixed it's all okay

**Comment by Exxion** [comment]
Worry not
Whop whop is an exclamation of excitement

**Comment by Exxion** [comment]
Wow, pixel movement already just barely begins to work with only this
But I really mean just barely
You still have 2004-era jumpy movement and you cease to be able to ever move again if you bump into something that is not tile-aligned
But hey! You do actually end up between tiles properly

Taking bets on how long it takes me to give up on making pixel movement work

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -68,10 +68,10 @@
 		close()
 	return
 
-/obj/machinery/door/window/CanPass(atom/movable/mover, turf/target, height=1.5, air_group = 0)
+/obj/machinery/door/window/Cross(atom/movable/mover, turf/target, height=1.5, air_group = 0)
```
Comment (Probe1) [inline code/game/machinery/doors/windowdoor.dm:71]: Check issues against this

https://github.com/d3athrow/vgstation13/issues/9862
https://github.com/d3athrow/vgstation13/issues/9861

**Pull request merged**